src/Compilers/CSharp/Portable/Symbols/Extensions/SourceExtensionImplementationMethodSymbol.cs

```diff
@@ -78,10 +81,38 @@ internal sealed override void AddSynthesizedAttributes(PEModuleBuilder moduleBui
                 }
             }
 
-            base.AddSynthesizedAttributes(moduleBuilder, ref attributes);
             SourceMethodSymbol.AddSynthesizedAttributes(this, moduleBuilder, ref attributes);
         }
 
+        internal override void AddSynthesizedReturnTypeAttributes(PEModuleBuilder moduleBuilder, ref ArrayBuilder<CSharpAttributeData> attributes)
+        {
+            // Copy NotNull/MaybeNull attribute from the property onto the implementation getter
+            if (_originalMethod is SourcePropertyAccessorSymbol { MethodKind: MethodKind.PropertyGet, AssociatedSymbol: SourcePropertySymbolBase extensionProperty })
+            {
+                foreach (CSharpAttributeData attr in extensionProperty.GetAttributes())
+                {
+                    if (attr.IsTargetAttribute(AttributeDescription.NotNullAttribute)
```

Comment:
>if (attr.IsTargetAttribute(AttributeDescription.NotNullAttribute) [](http://example.com/codeflow?start=20&length=65)

Do we have tests that explicitly verify presence of these attributes in metadata?

Reply:
See `Nullability_PropertyAccess_16`